File: 2350-shortest-impossible-sequence-of-rolls/2350-shortest-impossible-sequence-of-rolls.java
class Solution {
    public int shortestSequence(int[] rolls, int k) {
        HashSet<Integer> s = new HashSet<>();
        int res = 1;
        for(int i : rolls) {
            s.add(i);
            if(s.size() == k) {
                s.clear(); ++res;
            }
        }
        return res;
    }
}

File: 2350-shortest-impossible-sequence-of-rolls/2350-shortest-impossible-sequence-of-rolls.js
/**
 * @param {number[]} rolls
 * @param {number} k
 * @return {number}
 */
var shortestSequence = function(rolls, k) {
    const s = new Set();
    let res = 1;
    for (let i of rolls) {
        s.add(i);
        if (s.size === k) {
            s.clear();
            res++;
        }
    }
    return res;
};